step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepository.java

```diff
@@ -136,17 +126,49 @@ private PackageExecutionContext getOrRestorePackageExecutionContext(Map<String,
             );
 
             // create single automation package in isolated manager
-            try (FileInputStream fis = new FileInputStream(apFile)) {
-                inMemoryPackageManager.createAutomationPackage(fis, apFile.getName(), enricher, predicate);
+            try (FileInputStream fis = new FileInputStream(apFile.getFile())) {
+                inMemoryPackageManager.createAutomationPackage(fis, apFile.getFile().getName(), enricher, predicate);
             } catch (IOException e) {
-                throw new AutomationPackageManagerException("Cannot read the AP file: " + apFile.getName());
+                throw new AutomationPackageManagerException("Cannot read the AP file: " + apFile.getFile().getName());
             }
 
             return new PackageExecutionContext(contextId, inMemoryPackageManager, false);
         }
         return current;
     }
 
+    private AutomationPackageFile restoreApFile(String contextId, String apName, Map<String, String> repositoryParameters) {
```

Comment:
I would prefer to have a class extending the IsolatedAutomationPackageRepository for the antifactory case (maybe IsolatedAutomationPackageArtefactoryRepository ) where we overwrite getApFileForExecution and getApFileForExecution. You could have an instance of each in AutomationPackageExecutor and just select the correct instance at the beginning of runInIsolation whether the InputStream is null or not. I think the implementation would be cleaner.